hello-data-portal/hello-data-portal-api/src/main/java/ch/bedag/dap/hellodata/portal/superset/service/DashboardService.java

```diff
@@ -200,4 +214,45 @@ private void validateUserAllowedToUpdate(String instanceName, int subsystemId) {
             throw new ResponseStatusException(HttpStatus.EXPECTATION_FAILED, "User with specified email not found in subsystem");
         }
     }
+
+    public void uploadDashboardsFile(MultipartFile file, String contextKey) {
+        try (InputStream inputStream = file.getInputStream()) {
+            String fileName = file.getName();
+
+            String supersetInstanceName = metaInfoResourceService.findSupersetInstanceNameByContextKey(contextKey);
+            String subject = SlugifyUtil.slugify(supersetInstanceName + RequestReplySubject.UPLOAD_DASHBOARDS_FILE.getSubject());
+            String binaryFileId = UUID.randomUUID().toString();
+
+            int chunkNumber = 0;
+            int bytesRead;
+            byte[] buffer = new byte[file.getSize() > HALF_MB ? HALF_MB : (int) file.getSize()];
+            int bytesReadTotal = 0;
+            while ((bytesRead = inputStream.read(buffer)) != -1) {
+                bytesReadTotal += bytesRead;
+                boolean isLastChunk = bytesReadTotal == file.getSize();
+                log.info("Sending file to superset, bytes read total: {}, file size: {}", bytesReadTotal, file.getSize());
+                DashboardUpload dashboardUpload = new DashboardUpload(buffer, ++chunkNumber, fileName, binaryFileId, isLastChunk, file.getSize());
```

Comment:
Just sth minor, but will there be a chunk with number 0 like this?

Reply:
no, why?

Reply:
ah, ok. it's just because normally we use zero-based indexing